articles/121_realtime_proposal.md

```diff
@@ -0,0 +1,225 @@
+---
+layout: default
+title: Proposal for Implementation of Real-time Systems in ROS 2
+permalink: articles/realtime_proposal.html
+abstract: Proposal for a test-driven approach to the real-time performance requirement in ROS 2.
+published: true 
+author: Jackie Kay
+---
+
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+## Requirements and Implementation of Real-Time Systems
+
+### System requirements
+
+The requirements of a real-time system vary depending on the use case.
+At its core, the real-time requirement of a system has two components:
+
+- Latency
+
+  - Update period
+  - Determinism (measured by jitter)
```

Comment:
Just as a point of language - is this an appropriate use of "determinism"? This word invokes the idea of a philosophy to me, and the related word deterministic to me is fairly binary: It is either deterministic or not, there is no degree to which something can be deterministic or not. See:

- http://www.merriam-webster.com/dictionary/determinism

> determinism (1) philosophy : the belief that all events are caused by things that happened before them and that people have no real ability to make choices or control what happens

- https://en.wikipedia.org/wiki/Deterministic_algorithm

> ... given a particular input, will **always** produce the same output ...

I'm not sure what a better term would be, but predictable (predictability) seems like a candidate to me. The idea being:

- A hard realtime system's jitter is absolutely predictable therefore it is deterministic (guaranteed or proven).
- A soft realtime system's jitter is reasonably predictable (within a statistical bound).
 - At this point it is more of a variable quality of the system rather than a characteristic.
- A firm realtime system can adjust its prediction (requirement) for jitter based on failures?

Sorry to be so pedantic, but I think it will help us to use precise language where possible.

To be clear, I think your descriptions of those terms are spot on, I'm proposing we not use the word "determinism".